test_schedule.py

```diff
@@ -52,18 +52,80 @@ def setUp(self):
+            job_instance.week
+        with self.assertRaises(IntervalError):
+            job_instance.monday
+        with self.assertRaises(IntervalError):
+            job_instance.tuesday
+        with self.assertRaises(IntervalError):
+            job_instance.wednesday
+        with self.assertRaises(IntervalError):
+            job_instance.thursday
+        with self.assertRaises(IntervalError):
+            job_instance.friday
+        with self.assertRaises(IntervalError):
+            job_instance.saturday
+        with self.assertRaises(IntervalError):
+            job_instance.sunday
+
+        # test an invalid unit
+        job_instance.unit = "foo"
+        self.assertRaises(ScheduleValueError, job_instance.at, "1:0:0")
+        self.assertRaises(ScheduleValueError, job_instance._schedule_next_run)
+
+        # test start day exists but unit is not 'weeks'
+        job_instance.unit = "days"
+        job_instance.start_day = 1
+        self.assertRaises(ScheduleValueError, job_instance._schedule_next_run)
+
+        # test weeks with an invalid start day
+        job_instance.unit = "weeks"
+        job_instance.start_day = "bar"
+        self.assertRaises(ScheduleValueError, job_instance._schedule_next_run)
+
+        # test a valid unit with invalid hours/minutes/seconds
+        job_instance.unit = "days"
+        self.assertRaises(ScheduleValueError, job_instance.at, "25:00:00")
+        self.assertRaises(ScheduleValueError, job_instance.at, "00:61:00")
+        self.assertRaises(ScheduleValueError, job_instance.at, "00:00:61")
+
+        # test invalid time format
+        self.assertRaises(ScheduleValueError, job_instance.at, "25:0:0")
+        self.assertRaises(ScheduleValueError, job_instance.at, "0:61:0")
+        self.assertRaises(ScheduleValueError, job_instance.at, "0:0:61")
+
+        # test (very specific) seconds with unspecified start_day
+        job_instance.unit = "seconds"
+        job_instance.at_time = datetime.datetime.now()
+        job_instance.start_day = None
+        self.assertRaises(ScheduleValueError, job_instance._schedule_next_run)
+
+        # test self.latest >= self.interval
+        job_instance.latest = 3
+        self.assertRaises(ScheduleValueError, job_instance._schedule_next_run)
 
     def test_singular_time_units_match_plural_units(self):
-        assert every().second.unit == every().seconds.unit
-        assert every().minute.unit == every().minutes.unit
-        assert every().hour.unit == every().hours.unit
-        assert every().day.unit == every().days.unit
-        assert every().week.unit == every().weeks.unit
+        self.assertEqual(every().second.unit, every().seconds.unit)
+        self.assertEqual(every().minute.unit, every().minutes.unit)
+        self.assertEqual(every().hour.unit, every().hours.unit)
+        self.assertEqual(every().day.unit, every().days.unit)
+        self.assertEqual(every().week.unit, every().weeks.unit)
```

Comment:
Please don't switch those assertions because we're using PyTest which recommends the plain `assert` style. (Same applies to the other tests) Thx

Reply:
Sorry! I've reverted all of those changes.